core/src/main/java/org/elasticsearch/action/support/replication/TransportReplicationAction.java

```diff
@@ -314,23 +363,17 @@ protected void responseWithFailure(Throwable t) {
 
         @Override
         protected void doRun() throws Exception {
-            try (Releasable shardReference = getIndexShardOperationsCounter(request.internalShardId)) {
-                shardOperationOnReplica(request.internalShardId, request);
+            assert request.shardId() != null : "request shardId must be set";
+            try (Releasable shardReference = getIndexShardOperationsCounter(request.shardId())) {
```

Comment:
Nit: Can we rename `shardReference` to `ignored`? I don't know about Eclipse, but this will at least prevent IntelliJ from issuing unused variable warnings.